packages/terra-overlay/src/LoadingOverlay.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import IconSpinner from 'terra-icon/lib/icon/IconSpinner';
+import 'terra-base/lib/baseStyles';
+
+import Overlay from './Overlay';
+import styles from './Overlay.scss';
+
+const cx = classNames.bind(styles);
+
+const BackgroundStyles = Overlay.Opts.BackgroundStyles;
+
+const propTypes = {
+ /**
+ * The message to be displayed within the overlay.
+ */
+  message: PropTypes.string,
+  /**
+  * Indicates if the icon spinner should be animated.
+  */
+  isAnimated: PropTypes.bool,
+  /**
+  * The visual theme to be applied to the overlay background. Accepts 'light', 'dark', and 'clear' or BackgroundStyles.LIGHT, BackgroundStyles.DARK, and BackgroundStyles.CLEAR.
+  */
+  backgroundStyle: PropTypes.oneOf(['light', 'dark', 'clear', BackgroundStyles]),
+  /**
+  * Indicates if the overlay is open.
+  */
+  isOpen: PropTypes.bool,
+  /**
+  * Indicates if the overlay is relative to the triggering container.
+  */
+  isRelativeToContainer: PropTypes.bool,
+};
+
+const defaultProps = {
+  isAnimated: false,
+  isOpen: false,
+  backgroundStyle: BackgroundStyles.LIGHT,
+  isRelativeToContainer: false,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+  /* eslint-enable consistent-return */
+};
+
+const LoadingOverlay = ({
+  message,
+  isAnimated,
+  ...customProps }, {
+  intl,
+}) => {
+  // eslint-disable-next-line no-param-reassign
+  delete customProps.onRequestClose;
```

Comment:
Here are the two comments that lead to this implementation: [addressing props](https://github.com/cerner/terra-core/pull/510/files/d8868b58a2cde3bed80633f43ac322b96215bdbb#diff-eb98e157fb026e5939002705acee7472R40)
and [addressing delete](https://github.com/cerner/terra-core/pull/510/files/d8868b58a2cde3bed80633f43ac322b96215bdbb#diff-eb98e157fb026e5939002705acee7472R45).